### PR TITLE
[BACKLOG-15284] - ServiceObserver logic was not correct for slow writ…

### PR DESCRIPTION
**Pull request by kurtwalker**
…ing transformations

### CODE REVIEW

**Comment by kurtwalker** [comment]
@pentaho/ackbar

File: src/main/java/org/pentaho/di/trans/dataservice/optimization/cache/ServiceObserver.java [hunk 1]
```diff
@@ -68,7 +68,11 @@ public ServiceObserver( DataServiceExecutor executor ) {
         if ( isRunning ) {
           latch = new CountDownLatch( 1 );
           try {
-            latch.await( 1, TimeUnit.SECONDS );
+            while ( !latch.await( 1, TimeUnit.SECONDS ) ) {
+              if ( !isRunning ) {
```
Comment (nantunes) [inline src/main/java/org/pentaho/di/trans/dataservice/optimization/cache/ServiceObserver.java]: What's the difference between this `if` returning or falling to the`return` bellow?

Comment (nantunes) [inline src/main/java/org/pentaho/di/trans/dataservice/optimization/cache/ServiceObserver.java]: Sorry, my bad. Misread the code. 😌

Comment (nantunes) [inline src/main/java/org/pentaho/di/trans/dataservice/optimization/cache/ServiceObserver.java]: Of course you could just `break` but I was questioning the `if` itself. It's ok.

**Pull request merged**